### ISSUE TITLE
Mappings are wrong on Apple platform

### ISSUE THREAD

**Issue by Midar** [milesj/emojibase#48]
I have been redirected here from http://github.com/vector-im/riot-web:

It seems the current mappings break Apple platforms:
For example, `:(` is ☹︎, which is not rendered on Apple platforms as an Emoji, while 🙁 is an Emoji.
`:D` is rendered as 😁, when it is more commonly rendered as 😀 (not really a breakage, just inconsistency).
Things like :thumbsup: result in 👍️, which is rendered as a thumb up followed by a white box, while 👍 renders correctly.

Basically, look at the entire list on an iOS or mac device, and most of them are broken.

**Comment by milesj** [comment]
@Midar This really depends on what version of emojibase is being used, and that iOS/Safari versions. Do you have more information?

**Comment by Midar** [comment]
Latest version for everything. Also even happens with Safari Technology Preview. Firefox, OTOH, renders everything correct when a different Emoji font is used.

The ☹︎ seems to be universally wrong outside of Firefox, though.

**Comment by milesj** [comment]
@Midar Well emojibase-data v5 came out last week (emoji v13 support). Is it that version or v4?

There screenshots or anything else? Theres not much I can do to reproduce this on my end.

**Comment by Midar** [comment]
Here's two examples of what I mean:
<img width="73" alt="Bildschirmfoto 2020-03-21 um 17 16 28" src="https://user-images.githubusercontent.com/752237/77230888-dcfca500-6b97-11ea-902c-7002bbb0eb76.png">
<img width="69" alt="Bildschirmfoto 2020-03-21 um 17 16 55" src="https://user-images.githubusercontent.com/752237/77230892-e0902c00-6b97-11ea-81fa-c69183dcbe70.png">

**Comment by milesj** [comment]
Do you know where that happens in the code? I can check their implementation.

**Comment by Midar** [comment]
Unfortunately I don't. Let's loop someone in: @t3chguy

**Comment by milesj** [comment]
I figured out the `:(` one, but the thumbs up I'm still not sure.

**Comment by t3chguy** [comment]
So riot-web is currently using v4, I will get that updated to see if the mappings change.

**Comment by milesj** [comment]
This should fix the emoticons: https://github.com/milesj/emojibase/commit/a39d1a83946d2420342a54165383e252966510dd

Need more info for the thumbs up.

**Comment by milesj** [comment]
Actually, I found the root problem. So heres the source data for `2639`.

<img width="302" alt="Screen Shot 2020-03-21 at 12 04 33" src="https://user-images.githubusercontent.com/143744/77234498-30f19480-6b6c-11ea-8d15-8857e0231dcb.png">

The issue is that the default presentation type is text (`0`) instead of emoji (`1`), and that's why the non-emoji character is being used.

I'm assuming the `compact` dataset is being used? Maybe I'll just always force it to emoji since that's typically what people want.

**Comment by t3chguy** [comment]
Yes the project uses the compact dataset

**Comment by milesj** [comment]
Ok, I'm gonna do a retro breaking change on v5, so compact will always use emojis: https://github.com/milesj/emojibase/commit/14d8b9794d88dbfa1e765d1bf6e677654172e540#diff-8128864d03457e2abb445b92204a88feL25

When you update to v5, let me know if this is still an issue.

**Comment by t3chguy** [comment]
@Midar could you re-test on riot.im/develop which should be on emojibase-data@5 and emojibase-regex@4

Thanks

**Comment by Midar** [comment]
The sad Emoji now has the same white box on the right of it liks others like 👍 already had.

<img width="35" src="https://user-images.githubusercontent.com/752237/78614569-2a9f3000-786f-11ea-92bf-495de80122ea.png">

**Comment by milesj** [comment]
Hrmm, really can't help more unless I know which emoji hexcode is being rendered, and how, and what the actual dataset for that item looks like.

**Comment by Midar** [comment]
Finally got to test this now that there's a new Riot release:

<img width="110" src="https://user-images.githubusercontent.com/752237/81349880-2a3dc300-90c1-11ea-9cee-dc9470cbdf06.png">

So it seems :) is now fixed, but the other problems remain.

**Comment by t3chguy** [comment]
Sorry, should have updated here. Riot has had to roll back to emojibase-data@4 due to us not having the time to create an updated build of Twemoji to ship with the app for emoji 13 support. So how the app is now should match the state of this issue when it was created. :(

**Comment by ajbura** [comment]
Hi, I am using emojibase-data@6.2.0 and in app.cinny.in and having similar issues. Ping me if you need more info.

![Cinny-emoji-issue](https://user-images.githubusercontent.com/32841439/147848821-50ab4bb3-489a-4ee8-bb33-8bdbe584f7ce.png)

**Comment by slugalisk** [comment]
This is still an issue using the `compact` dataset from `emojibase-data@7.0.1`. There are 342 affected entries https://gist.github.com/slugalisk/750e211687215a1dc7be528ed3004e91. For some reason the `unicode` field is generated with an extra `FEOF` even though the `hexcode` looks right.
